sdk/monitor/opentelemetry-exporter-azuremonitor/src/main/java/com/azure/opentelemetry/exporter/azuremonitor/AzureMonitorExporter.java

```diff
@@ -0,0 +1,783 @@
+        exceptionData.getProperties().put("Logger Message", message);
+        setProperties(exceptionData.getProperties(), timeEpochNanos, level, loggerName, attributes);
+        telemetryItem.setSampleRate(samplingPercentage.floatValue());
+        telemetryItems.add(telemetryItem);
+    }
+
+    private static List<TelemetryExceptionDetails> minimalParse(String errorStack) {
+        TelemetryExceptionDetails details = new TelemetryExceptionDetails();
+        String line = errorStack.split("\n")[0];
+        int index = line.indexOf(": ");
+
+        if (index != -1) {
+            details.setTypeName(line.substring(0, index));
+            details.setMessage(line.substring(index + 2));
+        } else {
+            details.setTypeName(line);
+        }
+        details.setStack(errorStack);
+        return Arrays.asList(details);
+    }
+
+    private void exportRemoteDependency(String stdComponent, SpanData span, boolean inProc, List<TelemetryItem> telemetryItems) {
+        TelemetryItem telemetryItem = new TelemetryItem();
+        RemoteDependencyData remoteDependencyData = new RemoteDependencyData();
+        MonitorBase monitorBase = new MonitorBase();
+
+        telemetryItem.setTags(new HashMap<>());
+        telemetryItem.setName(telemetryItemNamePrefix + "RemoteDependency");
+        telemetryItem.setVersion(1);
+        telemetryItem.setInstrumentationKey(instrumentationKey);
+        telemetryItem.setData(monitorBase);
+
+        remoteDependencyData.setProperties(new HashMap<>());
+        remoteDependencyData.setVersion(2);
+        monitorBase.setBaseType("RemoteDependencyData");
+        monitorBase.setBaseData(remoteDependencyData);
+
+        addLinks(remoteDependencyData.getProperties(), span.getLinks());
+        remoteDependencyData.setName(span.getName());
+
+        span.getInstrumentationLibraryInfo().getName();
+
+        Map<String, AttributeValue> attributes = getAttributesCopy(span.getAttributes());
+
+        if (inProc) {
+            remoteDependencyData.setType("InProc");
+        } else {
+            if (attributes.containsKey("http.method")) {
+                applyHttpRequestSpan(attributes, remoteDependencyData);
+            } else if (attributes.containsKey(SemanticAttributes.DB_SYSTEM.key())) {
+                applyDatabaseQuerySpan(attributes, remoteDependencyData, stdComponent);
+            } else if (span.getName().equals("EventHubs.send")) {
+                // TODO eventhubs should use CLIENT instead of PRODUCER
+                // TODO eventhubs should add links to messages?
+                remoteDependencyData.setType("Microsoft.EventHub");
+                String peerAddress = removeAttributeString(attributes, "peer.address");
+                String destination = removeAttributeString(attributes, "message_bus.destination");
+                remoteDependencyData.setTarget(peerAddress + "/" + destination);
+            } else if (span.getName().equals("EventHubs.message")) {
```

Comment:
Do you have a constant's class somewhere? I see some of these are used multiple times and could be fragile if a replacement happens in one place and not another.

Reply:
Yes, we'll have a constants class in a follow-up PR.